plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2581,7 +2540,7 @@ void producer_plugin_impl::schedule_production_loop() {
       chain::controller& chain = chain_plug->chain();
       fc_dlog(_log, "Speculative Block Created; Scheduling Speculative/Production Change");
       EOS_ASSERT( chain.is_building_block(), missing_pending_block_state, "speculating without pending_block_state" );
-      schedule_delayed_production_loop(weak_from_this(), calculate_producer_wake_up_time(chain.pending_block_time()));
+      schedule_delayed_production_loop(weak_from_this(), calculate_producer_wake_up_time(chain.pending_block_timestamp()));
```

Comment:
I have tried to mark the conversion constructor and operator in `block_timestamp_type` explicit. However, it causes too many compiler errors and requires some effort to analyze each of them. I don't think it's in the scope of the PR. It might be worthwhile to create a new issue to pursue this direction.